crowbar_framework/app/models/service_object.rb

```diff
@@ -1884,6 +1918,11 @@ def skip_unchanged_nodes(elements, old_role, role)
     cleaned_elements
   end
 
+  # if this is true, node should not be skipped even though it's in crowbar_upgrade state
+  def crowbar_upgrade_needs_to_run(node)
```

Comment:
Nit: I think this name is somewhat misleading. I'd suggest to reverse the logic here and call the method something like `upgrade_postponed?`

Also the `node` parameter is unused. Is it actually needed?

Reply:
True about `node` and comments. 
`upgrade_postponed?` - OK, I can change it, but then I'll have to negate the output in the check later which is somewhat less nice.